content/articles/jaspersoft-reports/index.md

```diff
@@ -0,0 +1,720 @@
+        String uploadDir = StringUtils.cleanPath("./generated-reports");
+        Path uploadPath = Paths.get(uploadDir);
+        if (!Files.exists(uploadPath)){
+            Files.createDirectories(uploadPath);
+        }
+        //generate the report and save it in the just created folder
+        if (fileFormat.equalsIgnoreCase("pdf")){
+            JasperExportManager.exportReportToPdfFile(
+                    jasperPrint, uploadPath+fileName
+            );
+        }
+
+        return uploadPath;
+    }
+
+    private String getPdfFileLink(String uploadPath){
+        return uploadPath+"/"+"products.pdf";
+    }
+
+    @Override
+    public String generateReport(LocalDate localDate, String fileFormat){
+
+        return null;
+
+    }
+    
+    @Override
+    public List<Product> findAllProducts() {
+        return null;
+    }
+}
+``` 
+### Create a resource handler to display the generated pdf
+Create a class named `AppWebConfig` that implements `MvcConfigurer` and override the `addResourceHandlers()` method. Then, create a method named `uploadPath()` that returns the absolute path of our specified folder. We indicate using the handler that the application should allow requests to our pdf links using the string `/generated-reports/**`. We then pass the absolute path returned by the method as the resource location of our generated report. 
+
+```java
+import org.springframework.context.annotation.Configuration;
+import org.springframework.web.servlet.config.annotation.ResourceHandlerRegistry;
+import org.springframework.web.servlet.config.annotation.WebMvcConfigurer;
+
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+@Configuration
+public class AppWebConfig implements WebMvcConfigurer {
+    @Override
+    public void addResourceHandlers(ResourceHandlerRegistry registry) {
+        String reportPath = uploadPath("./generated-reports");
+
+        registry.addResourceHandler("/generated-reports/**")
+                .addResourceLocations("file:/"+reportPath+"/");
+    }
+
+    private String uploadPath(String directory){
+        Path uploadDirPath = Paths.get(directory);
+        return uploadDirPath.toFile().getAbsolutePath();
+    }
+}
+```
+
+### Generate a report with the filtered products
+This functionality will be implemented using the `generateReport()` method in the `ReportServiceImpl` class. This method accepts two parameters of type `LocalDate` and `String`. Since this method will be responsible for searching products by date, we inject the `ProductsRepository` bean. We use it to search for products created on that specific day using the `findAllByCreatedAt()` method.
+
+Create a string containing the resource location of our source file. i.e, `products.jrxml`. Call the `getJasperPrint()` method, and pass the collection of products retrieved from the database as the first parameter and the above string as the second parameter.
```

Comment:
> Create a string containing the resource location of our source file. i.e, `products.jrxml`. Call the `getJasperPrint()` method, and pass the collection of products retrieved from the database as the first parameter and the above string as the second parameter.
> 
> Create a string containing the name of the pdf report to be generated. i.e `products.pdf`.
> 
> Call the `getUploadPath()` method and the parameter string provided by the `generateReport()` as the file format,` JasperPrint` returned by the `getJasperPrint()` method and the above string of our file name.
> 
> Finally, return the `getFilePdfLink()` method and pass the value returned by the `getUploadPath()` as a string.

It doesn't make sense to separate these into separate paragraphs. It makes it seem like they are unrelated in some way when in reality, these are steps in the logic of the method talked about in the paragraph before this. When editing this, I wasn't even sure at first that this code belonged in that same method or where it even belonged because of this. Find a way to group this together so it's more obvious that they are related.